File: Module.Persist.TPM/Model/History/HistoricalBudget.cs
using Core.History;
using Module.Persist.TPM.Model.TPM;
using System;

namespace Module.Persist.TPM.Model.History
{
    [AssociatedWith(typeof(Budget))]
    public class HistoricalBudget : BaseHistoricalEntity<System.Guid>
    {
        public string Name { get; set; }
    }
}

File: Module.Frontend.TPM/Client/model/tpm/budget/HistoricalBudget.js
Ext.define('App.model.tpm.budget.HistoricalBudget', {
    extend: 'Ext.data.Model',
    idProperty: '_Id',
    breezeEntityType: 'Budget',
    fields: [
        { name: '_Id', type: 'string', hidden: true },
        { name: '_ObjectId', hidden: true },
        { name: '_User', type: 'string', isDefault: true },
        { name: '_Role', type: 'string', isDefault: true },
        { name: '_EditDate', type: 'date', isDefault: true },
        { name: '_Operation', type: 'string', isDefault: true },
        { name: 'Name', type: 'string', hidden: false, isDefault: true }
    ],
    proxy: {
        type: 'breeze',
        resourceName: 'HistoricalBudgets',
        reader: {
            type: 'json',
            totalProperty: 'inlineCount',
            root: 'results'
        }
    }
});
